Vagrantfile

```diff
@@ -0,0 +1,129 @@
+  #   vb.memory = "1024"
+  # end
+  #
+  # View the documentation for the provider you are using for more
+  # information on available options.
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Ansible, Chef, Docker, Puppet and Salt are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  # config.vm.provision "shell", inline: <<-SHELL
+  #   apt-get update
+  #   apt-get install -y apache2
+  # SHELL
+  
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL
+	export PROFILE=/home/vagrant/.profile
+    export INSTALL_ROOT="/home/vagrant"
+    cd ${INSTALL_ROOT}
+	    sudo apt-get update
+    # Install pyenv prerequisites
+    export BUILD_LIBS="build-essential"
+    export BUILD_LIBS="${BUILD_LIBS} libssl-dev"
+    export BUILD_LIBS="${BUILD_LIBS} zlib1g-dev"
+    export BUILD_LIBS="${BUILD_LIBS} libbz2-dev"
+    export BUILD_LIBS="${BUILD_LIBS} libreadline-dev"
+    export BUILD_LIBS="${BUILD_LIBS} libsqlite3-dev"
+    export BUILD_LIBS="${BUILD_LIBS} wget"
+    export BUILD_LIBS="${BUILD_LIBS} curl"
+    export BUILD_LIBS="${BUILD_LIBS} llvm"
+    export BUILD_LIBS="${BUILD_LIBS} libncursesw5-dev"
+    export BUILD_LIBS="${BUILD_LIBS} xz-utils"
+    export BUILD_LIBS="${BUILD_LIBS} tk-dev"
+    export BUILD_LIBS="${BUILD_LIBS} libxml2-dev"
+    export BUILD_LIBS="${BUILD_LIBS} libxmlsec1-dev"
+    export BUILD_LIBS="${BUILD_LIBS} libffi-dev"
+    export BUILD_LIBS="${BUILD_LIBS} liblzma-dev"
+    sudo apt-get --yes install make ${BUILD_LIBS}
+    # Prep pyenv env
+    export VAGRANT_INSTALL=${INSTALL_ROOT}/.pyenv
+    # Install pyenv
+    git clone https://github.com/pyenv/pyenv.git ${VAGRANT_INSTALL}
+    export PATH=${VAGRANT_INSTALL}/bin:${PATH}
+    # Update ~/.profile
+    echo "" >> ${PROFILE}
+    echo "# set Vagrant env vars" >> ${PROFILE}
+    echo "export VAGRANT_INSTALL=${VAGRANT_INSTALL}" >> ${PROFILE}
+    echo "export PATH=${VAGRANT_INSTALL}/bin:${PATH}" >> ${PROFILE}
+    echo "`pyenv init --path`" >> ${PROFILE}
+    # Install Python 3.8.5
+    export TARGET_PYTHON_VERSION="3.8.5"
+    pyenv install ${TARGET_PYTHON_VERSION}
+    pyenv global ${TARGET_PYTHON_VERSION}
+    # Update pip
+    echo "Updating pip"
+    ${VAGRANT_INSTALL}/versions/${TARGET_PYTHON_VERSION}/bin/python3.8 -m pip install --upgrade pip
+    # Install poetry
+    echo "Installing poetry"
+    curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+  SHELL
+  
+  config.trigger.after :up do |trigger|
+	 trigger.name = "Launching App"
+	 trigger.info = "Running the TODO app setup script"
+	 #trigger.run_remote = {privileged: false, inline: "cd /vagrant && ./setup_on_vagrant.sh && ./run_gunicorn.sh"}
+	 trigger.run_remote = {privileged: false, inline: "
+	 # Install dependencies and launch
+	 cd /vagrant && poetry install && cd /vagrant/todo_app && nohup poetry run flask run --host=0.0.0.0 > logs.txt 2>&1 &
+	 "}
```

Comment:
Your trigger never executed all of these commands for me - does it work for you? I wonder whether it's being caught by [this limitation on chaining commands](https://stackoverflow.com/questions/55167743/how-can-i-run-multiple-commands-in-a-vagrant-trigger) meaning that only the first (changing directory) was being executed
Switching it out for something like:
```
	 trigger.run_remote = {privileged: false, inline: <<-SHELL
      cd /vagrant
      poetry install
      cd /vagrant/todo_app
      nohup poetry run flask run --host=0.0.0.0 > logs.txt 2>&1 &
    SHELL
```
Fixed that up for me

Reply:
You're right, it didn't work for me! But I was a bit lost...
Thanks so much for your help
(I had some troubles running the application because I forgot to setup my .env file)
I still don't seem to get any logs from python logging which makes it difficult to investigate issues...